Documentation/ApiOverview/Events/Events/Linkvalidator/BeforeRecordIsAnalyzedEvent.rst

```diff
@@ -10,6 +10,144 @@ BeforeRecordIsAnalyzedEvent
 Event that is fired to modify results (= add results) or modify the record
 before the :doc:`LinkValidator <ext_linkvalidator:Index>` analyzes the record.
 
+Example
+=======
+
+In this example we are checking if there are external links containing the url
+of the project itself as editors tend to set external links on internal pages
+at times.
+
+The following code has to be put in a custom
+:ref:`minimal extension <extension-minimal>`.
+
+Create a class that works as event listener. This class does not implement or
+extend any class. It has to provide a method that accepts an event of type
+:php:`TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent`.  By default
+the method is called :php:`handleEvent`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace Vendor\MyExtension\EventListener\Linkvalidator;
+
+    use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+
+    class BeforeRecordIsAnalyzedEventListener
+    {
+
+        public function handleEvent(BeforeRecordIsAnalyzedEvent $event): void
+        {
+            // do something
+        }
+    }
+
+The listener must then be registered in the extensions :php:`Services.yaml`:
+
+..  code-block:: yaml
+    :caption: EXT:my_extension/Configuration/Services.yaml
+
+    services:
+        _defaults:
+            autowire: true
+            autoconfigure: true
+            public: false
+
+        Vendor\MyExtension\:
+            resource: '../Classes/*'
+            exclude: '../Classes/Domain/Model/*'
+
+        Vendor\MyExtension\EventListener\Linkvalidator\BeforeRecordIsAnalyzedEventListener:
+            tags:
+                - name: event.listener
+                  method: handleEvent
+                  identifier: 'myExtensionBeforeRecordIsAnalyzedEventListener'
+                  event: TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent
+
+For the implementation we need the :php:`BrokenLinkRepository` to register
+additional link errors and the :php:`SoftReferenceParserFactory` so we can
+automatically parse for links. These two classes have to be injected via
+:ref:`dependeny injection <Dependency-Injection>`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    use TYPO3\CMS\Core\DataHandling\SoftReference\SoftReferenceParserFactory;
+    use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+    use TYPO3\CMS\Linkvalidator\Repository\BrokenLinkRepository;
+
+    class BeforeRecordIsAnalyzedEventListener
```

Comment:
```suggestion
    final class BeforeRecordIsAnalyzedEventListener
```

Would prefer to define it final (opinionated). We also use this in other examples of events.